File: Source/JapaneseLanguageTools.Contracts/Models/Responses/Base/ExportResponse.Model.cs
using JapaneseLanguageTools.Contracts.Models.Blobs;

namespace JapaneseLanguageTools.Contracts.Models.Responses.Base;

public class ExportResponseModel
{
    public required ExportStatus Status { get; set; }

    public required ExportResponseMessageModel[] Messages { get; set; }

    public required BlobReferenceModel BlobReference { get; set; }
}

File: Source/JapaneseLanguageTools.ReactSpa/source/models/responses/base/ExportResponse.Model.ts
import BlobReferenceModel from "@/models/blobs/BlobReference.Model";
import ExportResponseMessageModel from "@/models/responses/base/ExportResponseMessage.Model";
import ExportStatus from "@/models/responses/base/ExportStatus";

export default interface ExportResponseModel {
  status: ExportStatus;
  messages: ExportResponseMessageModel[];
  blobReference: BlobReferenceModel;
}
